content/reference/v1beta1/repeatable.md

```diff
@@ -0,0 +1,278 @@
+---
+date: 2021-07-13
+linktitle: "Repeatable"
+title: Repeatable Config
+description: "Repeatable config generates copies of kubernetes manifest items."
+weight: 9
+---
+
+## Repeatable Items
+
+A repeatable config item copies a YAML array entry or YAML document for as many values as are provided. Any number of values can be added to a repeatable item to generate additional copies.
+
+**Repeatable Items presently only work for text and textarea types**
+
+To make an item repeatable, set `repeatable` to true
+```yaml
+    - name: ports
+      items:
+      - name: service_ports
+        title: Service Ports
+        type: text
+        repeatable: true
+```
+
+Repeatable items do not use the `default` or `value` fields, but instead a `valuesByGroup` field.
+`valuesByGroup` should have an entry for the parent Config Group name, with all default `key:value` pairs nested in the group.  1 default entry is required for the repeatable item.
+```yaml
+    valuesByGroup:
+      ports:
+        port-default-1: "80"
+```
+
+## Template Targets
+
+Repeatable items require at least 1 `template` to be provided.  The `template` defines a YAML target in the manifest to duplicate for each repeatable item.
+
+The entire YAML node at the target will be duplicated, including nested fields.
+
+The `yamlPath` field of the `template` must denote index position for arrays using square brackets.  For example, `spec.ports[0]` to select the first port entry for duplication.  All duplicate YAML will be appended to the final array in the `yamlPath`.
+
+`yamlPath` **must** end with an array.
+```yaml
+    template:
+    - apiVersion: v1
+      kind: Service
+      name: my-service
+      namespace: my-app
```

Comment:
Just the first three.  I've added the following line to "Template Targets": ```Required fields for a template target are `apiVersion`, `kind`, and `name`.```

Reply:
So what does specifying the namespace do? And what if I specified another field that isn't required?

Reply:
If you specify a namespace, it will only match a manifest with that namespace in the metadata.  For example, if you have `name: my-service` in a `data` namespace as well as in a `app` namespace, this will allow you to target a specific manifest.

Reply:
Are there any other fields a vendor can mention here that will have an effect? Since we now note that you have to provide `apiVersion`, `kind`, and `name`, we should mention anything else that can be put here too.

Reply:
Templates must have apiVersion, kind, and name.  They may have namespace and yamlPath.  Those are the only available properties today.  The KOTS linter is updated to enforce these rules as well, reducing the chances a vendor makes a mistake.

Reply:
`yamlPath` is obviously covered below. After mentioning the three required ones, I would say that you can specify the namespace in order to disambiguate if a yaml document with the same `apiVersion`, `kind`, and `name` exists for multiple namespaces. Just so someone knows that that is an option and why they'd need it. This can resolve too.

Reply:
I added a line to explain what namespaces can be used for to clarify :)

Reply:
Nice, good to resolve here too.